libminifi/CMakeLists.txt

```diff
@@ -74,6 +74,7 @@ target_link_libraries(core-minifi ${UUID_LIBRARIES} ${JSONCPP_LIB} yaml-cpp )
 find_package(ZLIB REQUIRED)
 include_directories(${ZLIB_INCLUDE_DIRS})
 
+target_link_libraries(core-minifi minifi-expression-language-extensions)
```

Comment:
I'm surprised to see this here. This means the extension cannot be fully disabled, so then we rely on the noop?

Reply:
Ah seems that way from the cmake.

Reply:
Yeah we rely on the noop. Design intent is for noop to be optimized out at compile time, but would have to double-check that's the case. Maybe it should be explicitly inlined so we inline nothing. I'm open to suggestions.

Reply:
Should be able to use enable_if to to completely eliminate the code and the function calls, but I'm not terribly worried. Cost will be low and can be changed in the future. 

It won't be completely optimized. With -O3 you'll get a stack setup, checks for the std::__throw_bad_function_call() and delete in the assembler. It's not the same as calling an empty function that does nothing, which can be optimized out, but in my opinion it's premature to concern ourselves over the cost.